### ISSUE TITLE
Slack error, I do not care about slack, how to disable

### ISSUE THREAD

**Issue by FortuneAlex** [Mixd/wp-deploy#70]
I keep getting a slack notification error, so it doesn't push... But I don't care about Slack and I ust want to disable it

vagrant@homestead:/vagrant/wp-deploy$ bundle exec cap production deploy
INFO Notifying Slack of deploy starting
(Backtrace restricted to imported tasks)
cap aborted!
:slack_url is not set

Tasks: TOP => deploy:starting => slack:notify_started
(See full trace by running task with --trace)
The deploy has failed with an error: :slack_url is not set
INFO Notifying Slack of deploy failed
/var/lib/gems/2.1.0/gems/capistrano-slackify-2.2.0/lib/capistrano/tasks/slackify.cap:49:in `block (3 levels) in <top (required)>': :slack_url is not set (RuntimeError)
        from /var/lib/gems/2.1.0/gems/capistrano-3.4.0/lib/capistrano/configuration.rb:41:in `call'
        from /var/lib/gems/2.1.0/gems/capistrano-3.4.0/lib/capistrano/configuration.rb:41:in `fetch'
        from /var/lib/gems/2.1.0/gems/capistrano-3.4.0/lib/capistrano/dsl/env.rb:10:in `fetch'
        from /var/lib/gems/2.1.0/gems/capistrano-slackify-2.2.0/lib/capistrano/tasks/slackify.cap:36:in `block (3 levels) in <top (required)>'
        from /var/lib/gems/2.1.0/gems/sshkit-1.7.1/lib/sshkit/backends/local.rb:15:in `instance_exec'
        from /var/lib/gems/2.1.0/gems/sshkit-1.7.1/lib/sshkit/backends/local.rb:15:in `run'
        from /var/lib/gems/2.1.0/gems/capistrano-3.4.0/lib/capistrano/dsl.rb:59:in `run_locally'
        from /var/lib/gems/2.1.0/gems/capistrano-slackify-2.2.0/lib/capistrano/tasks/slackify.cap:31:in `block (2 levels) in <top (required)>'
        from /var/lib/gems/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:240:in `call'
        from /var/lib/gems/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:240:in `block in execute'
        from /var/lib/gems/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:235:in `each'
        from /var/lib/gems/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:235:in `execute'
        from /var/lib/gems/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:179:in `block in invoke_with_call_chain'
        from /usr/lib/ruby/2.1.0/monitor.rb:211:in `mon_synchronize'
        from /var/lib/gems/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:172:in `invoke_with_call_chain'
        from /var/lib/gems/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:165:in `invoke'
        from /var/lib/gems/2.1.0/gems/capistrano-3.4.0/lib/capistrano/dsl/task_enhancements.rb:14:in `block in after'
        from /var/lib/gems/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:240:in `call'
        from /var/lib/gems/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:240:in `block in execute'
        from /var/lib/gems/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:235:in `each'
        from /var/lib/gems/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:235:in `execute'
        from /var/lib/gems/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:179:in `block in invoke_with_call_chain'
        from /usr/lib/ruby/2.1.0/monitor.rb:211:in `mon_synchronize'
        from /var/lib/gems/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:172:in `invoke_with_call_chain'
        from /var/lib/gems/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:165:in `invoke'
        from /var/lib/gems/2.1.0/gems/capistrano-3.4.0/lib/capistrano/dsl.rb:16:in `invoke'
        from /var/lib/gems/2.1.0/gems/capistrano-3.4.0/lib/capistrano/dsl/task_enhancements.rb:57:in `exit_deploy_because_of_exception'
        from /var/lib/gems/2.1.0/gems/capistrano-3.4.0/lib/capistrano/application.rb:78:in `exit_because_of_exception'
        from /var/lib/gems/2.1.0/gems/rake-10.4.2/lib/rake/application.rb:186:in `rescue in standard_exception_handling'
        from /var/lib/gems/2.1.0/gems/rake-10.4.2/lib/rake/application.rb:176:in `standard_exception_handling'
        from /var/lib/gems/2.1.0/gems/rake-10.4.2/lib/rake/application.rb:75:in `run'
        from /var/lib/gems/2.1.0/gems/capistrano-3.4.0/lib/capistrano/application.rb:15:in `run'
        from /var/lib/gems/2.1.0/gems/capistrano-3.4.0/bin/cap:3:in `<top (required)>'
        from /usr/local/bin/cap:23:in `load'
        from /usr/local/bin/cap:23:in `<main>'

Any ways to fix this? Thanks

**Comment by jyrkij** [comment]
Comment/remove [this line](https://github.com/Mixd/wp-deploy/blob/v1.2.0/Capfile#L15).

**Comment by lukewhitehouse** [comment]
Sorry for the error @FortuneAlex, we appreciate that a lot of users will not need Slack and we plan on making this an opt-in addon in the next major release.

For now, I'd recommend you do as @jyrkij says, that'll solve your problem. You can also remove [this line](https://github.com/Mixd/wp-deploy/blob/master/config/deploy.rb#L5) and [this file](https://github.com/Mixd/wp-deploy/blob/master/config/slack.rb) to fully remove slack from your project ( Thanks @jyrkij )

**Comment by FortuneAlex** [comment]
Thanks, I'll try out the solution tonight.

I juuuust realized that the title might have sounded rude, wasn't the intention.

**Comment by lukewhitehouse** [comment]
No worriees @FortuneAlex :) Let me know if you need a hand.

**Comment by mlwilkerson** [comment]
I'm having the same problem and I can't seem to disable this error message by simply disabling slack. I've removed the require like from deploy.rb and entirely removed the slack.rb file. Checked it in to my master branch. Did the same on my development branch. Pushing everything to my remote. I have no idea where the system is still possibly getting the slack stuff from such that it's still choking on it. 

Is it perhaps cached in some working directory somewhere that needs to be purged before trying again with the deploy script?
